src/main/java/com/service/UserService.java

```diff
@@ -0,0 +1,68 @@
+package com.service;
+
+import com.dao.UserDAO;
+import com.model.User;
+import lombok.Data;
+
+import javax.sql.DataSource;
+import java.util.Optional;
+
+@Data
```

Comment:
You don't need this annotation here because it's reasonable to apply to entites (@Data generates equals() and hashcode() also). What's the purpose?